komorebi/src/window.rs

```diff
@@ -159,6 +159,24 @@ impl Window {
         HWND(windows_api::as_ptr!(self.hwnd))
     }
 
+    pub fn move_to_area(&mut self, current_area: &Rect, target_area: &Rect) -> Result<()> {
```

Comment:
Can you ELI5 the difference between this and center?

Reply:
Basically the `center` function changes the size of the window to be half the target monitor's width and height and positions it in the center. The `move_to_area` keeps the size of the window and moves it to the target monitor to the "same" position as it was on the original monitor. If the resolutions of each monitor were different it uses the ratio to position the window as close to the same position as possible.

For example if monitor 1 is 1920x1080 and monitor 2 is 3840x2160. When moving from monitor 1 to 2 the ratio will be `2` when moving from monitor 2 to 1 the ratio will be `0.5`. So if window is on monitor 2 with the top left corner on position `x: 3000, y: 2000` relative to that screen top left corner, then it will position on monitor 1 on position `x:1500, y:1000` relative to monitor 1 top left corner. Which will correspond to a similar position, since on monitor 1 it is at `3000/3840*100=78.125%` of the screen (from left to right), on monitor 2 it will be at `1500/1920*100=78.125%` which will look the same. This is basically what the `x/y ratios` are for.

The case for the `x/y diffs` its easier to explain like this:
- Imagine that you have 3 monitors with 1000x500px.
- They are distributed from left to right
- monitor 1 coordinates
  ```rust
     top: 0,
     left: 0,
     right: 1000,
     bottom: 500,
  ```
- monitor 2 coordinates
  ```rust
     top: 0,
     left: 1000,
     right: 1000,
     bottom: 500,
  ```
- monitor 1 coordinates
  ```rust
     top: 0,
     left: 2000,
     right: 1000,
     bottom: 500,
  ```

Let's say window to move is on monitor 3 with its top left corner at relative position of `x:100, y:100` and with a size of `200x200`, so the rect for the window will be:
- monitor 1 coordinates
  ```rust
     top: 100,
     left: 2100,
     right: 200,
     bottom: 200,
  ```

When moving to monitor 1 we want the final position to be `x:100, y:100`, which in this case since it is the first monitor (at `0, 0`) it will actually correspond to `x:100, y:100`. The calculation is as follows:
`y_diff = target_y - current_y = 0 - 0 = 0`
`x_diff = target_x - current_x = 0 - 2000 = -2000`
- monitor 1 coordinates
  ```rust
     top: y_diff + window_y = 0 + 100 = 100,
     left: x_diff + window_x = -2000 + 2100 = 100,
     right: 200,
     bottom: 200,
  ```
Now that I've made this explanation I've noticed that I've not pushed the most recent change to fix an issue on this function, the `left` value should be similar to `top`, like this:
```rust
left: ((x_diff + current_rect.left) as f32 * x_ratio) as i32,
```
I have that change on my laptop and forgot to commit it. I'll do it tomorrow in the afternoon once I get some time...